src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/constrainedtransfer/ConstrainedBoardingSearch.java

```diff
@@ -105,9 +105,16 @@ public boolean transferExist(int targetStopPos) {
             int stopIndex
     ) {
         final Trip trip = tripSchedule.getOriginalTripTimes().getTrip();
-        return currentTransfers.stream()
-                .filter(t -> t.matchesSourcePoint(stopIndex, trip))
-                .collect(Collectors.toList());
+        // for performance reasons we use a for loop here as streams are much slower.
+        // i experimented with LinkedList and ArrayList and could see no appreciable difference in
+        // speed so went with ArrayList as it consumes less memory.
+        var result = new ArrayList<TransferForPattern>(50);
```

Comment:
Is there a difference if we leave the initial capacity out?

Reply:
So I experimented some more and it was actually faster to use a linked list after all. I can get faster and more reproducible results with it.